src/main/java/org/spongepowered/api/command/CommandManager.java

```diff
@@ -109,7 +109,77 @@
      * @throws IllegalArgumentException Thrown if {@code plugin} is not a
      *     plugin instance
      */
-    Optional<CommandMapping> register(Object plugin, CommandCallable callable, List<String> aliases, Function<List<String>, List<String>> callback);
+    Optional<CommandMapping> register(Object plugin, Command callable, List<String> aliases, Function<List<String>, List<String>> callback);
+
+    /**
+     * Register a given command using the given list of aliases.
+     *
+     * <p>If there is a conflict with one of the aliases (i.e. that alias
+     * is already assigned to another command), then the alias will be skipped.
+     * It is possible for there to be no alias to be available out of
+     * the provided list of aliases, which would mean that the command would not
+     * be assigned to any aliases.</p>
+     *
+     * <p>The first non-conflicted alias becomes the "primary alias."</p>
+     *
+     * @param pluginContainer A {@link PluginContainer}
+     * @param callable The command
+     * @param alias An array of aliases
+     * @return The registered command mapping, unless no aliases could be
+     *     registered
+     * @throws IllegalArgumentException Thrown if {@code plugin} is not a
+     *     plugin instance
+     */
+    Optional<CommandMapping> register(PluginContainer pluginContainer, Command callable, String... alias);
+
+    /**
+     * Register a given command using the given list of aliases.
+     *
+     * <p>If there is a conflict with one of the aliases (i.e. that alias
+     * is already assigned to another command), then the alias will be skipped.
+     * It is possible for there to be no alias to be available out of
+     * the provided list of aliases, which would mean that the command would
+     * not be assigned to any aliases.</p>
+     *
+     * <p>The first non-conflicted alias becomes the "primary alias."</p>
+     *
+     * @param pluginContainer A {@link PluginContainer}
+     * @param callable The command
+     * @param aliases A list of aliases
+     * @return The registered command mapping, unless no aliases could be
+     *     registered
+     * @throws IllegalArgumentException Thrown if {@code plugin} is not a
+     *     plugin instance
+     */
+    Optional<CommandMapping> register(PluginContainer pluginContainer, Command callable, List<String> aliases);
+
+    /**
+     * Register a given command using a given list of aliases.
```

Comment:
I'm fine with seeing it go.
This was added by @sk89q [back in 2014](https://github.com/SpongePowered/SpongeAPI/commit/de05a90402d4a76d1cf40d1ab32da1e2963e48c3), but as far as I know barely any plugins (if any at all) actually use this.